src/main/java/com/ibm/devops/connect/CRPipeline/UploadDeployment.java

```diff
@@ -0,0 +1,272 @@
+                } else if (cause instanceof UpstreamCause) {
+                    UpstreamCause upstreamCause = (UpstreamCause)cause;
+                    payload.put("by_user", "Upstream job \"" + upstreamCause.getUpstreamProject() + "\", build \"" + upstreamCause.getUpstreamBuild() + "\"");
+                } else if (cause instanceof RemoteCause) {
+                    RemoteCause remoteCause = (RemoteCause)cause;
+                    payload.put("requestor", remoteCause.getAddr());
+                } else {
+                    payload.put("requestor", cause.getShortDescription());
+                }
+            }
+        }
+        if (succeeded != null && !succeeded.equals("")) {
+            payload.put("result", succeeded.equals("true") ? "Success" : "Failed");
+        } else {
+            String computedStatus = "Failed";
+            Result buildResult = build.getResult();
+            if (buildResult == null || buildResult.equals(Result.SUCCESS)) {
+                computedStatus = "Success";
+            }
+            payload.put("result", computedStatus);
+        }
+        if (startTime != null && !startTime.equals("")) {
+            payload.put("start_time", Long.valueOf(startTime));
+        } else {
+            payload.put("start_time", build.getStartTimeInMillis());
+        }
+        if (endTime != null && !endTime.equals("")) {
+            payload.put("end_time", Long.valueOf(endTime));
+        } else {
+            payload.put("end_time", System.currentTimeMillis());
+        }
+
+
+        System.out.println("TEST payload: " + payload.toString(2));
+
+        if (this.debug != null && this.debug.toString().equals("true")) {
+            listener.getLogger().println("payload: " + payload.toString());
+        }
+
+        listener.getLogger().println("Uploading deployment \"" + payload.get("version_name") + "\" of \"" + payload.get("name") + "\" to UrbanCode Velocity...");
+        try {
+            String response = CloudPublisher.uploadDeployment(payload.toString());
+            System.out.println("TEST response: " + response);
+            JSONObject json = JSONObject.fromObject(response);
+            if (json.isEmpty() || !json.has("_id") || json.get("_id").equals("")) {
+                throw new RuntimeException("Did not receive successful response: " + response);
+            }
+            listener.getLogger().println("Successfully uploaded deployment to UrbanCode Velocity.");
+        } catch (Exception ex) {
+            listener.error("Error uploading deployment data: " + ex.getClass() + " - " + ex.getMessage());
+            build.setResult(Result.FAILURE);
+        }
+    }
+
+    @Extension
+    public static class UploadDeploymentDescriptor extends BuildStepDescriptor<Builder> {
+
+        public UploadDeploymentDescriptor() {
+            load();
+        }
+
+        @Override
+        public String getHelpFile() {
+            return "/plugin/ibm-ucdeploy-build-steps/publish.html";
```

Comment:
I think this should change... and probably for the other steps as well

Reply:
removed